### ISSUE TITLE
Audio continues after video buffer ends

### ISSUE THREAD

**Issue by Winterhuman** [FreeTubeApp/FreeTube#1557]
**Behavior of the program**
When the video buffer runs out, the video will often have more audio content than video content, this causes the last frame of the video buffer to be displayed while the audio continues for a few seconds after, but, when more buffer is loaded it continues from the end of the audio and misses out on the video content inbetween causing a visual jump between loaded buffers.

**Expected behavior**
Stop playing the video at the end of the video content's buffer (Unless audio-only mode is used) rather than the audio content's to make sure video content is never missed out when loading more buffer.

**To Reproduce**
1. Play a video and cut off the internet at some point.
2. The video stops, but, audio can still be heard for a few second after.
3. Reconnect the internet and see the video continue from the end of the audio.

**Environment Information:**
 - OS: Arch Linux
 - FreeTube version: 0.15.1 Beta
 - Installation Method: FlatHub
 - Primary API Used: Local API

**Additional context**
This "issue" is purely cosmetic and isn't really significant to the viewing experience; it's low priority if even an issue at all.

**Comment by RootBeerFan** [comment]
Still present in latest version.

**Comment by azurusnova** [comment]
I keep having this issue myself. Along with when the video buffer will run out and not catch up, I would end up having to hit CTRL+R to reboot the window, and sometimes it will pick up where it left off, other times (and most of the times) it will start from the beginning all over. 

This happens way too often with a LOT of videos. Really not sure what the fix is, but FreeTube is about the only way I can really watch youtube anymore. Not able to watch it in a browser because of a PulseAudio bug that crashes PulseAudio anytime audio is played from a browser. Linux Mint hasn't really gone out of it's way to update to a newer version to fix that issue, but that's just another problem entirely. 

Sorry for the little rant. 
Thank you for coming to my TedTalk. XD

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@azurusnova try using the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by RootBeerFan** [comment]
I discovered some possible solution to the problem.
I had the issue myself for quite some time, even on different platforms (win & linux) and tried many attempts myself over the course of lets say at least a year.

The following happend; On my Windows machine I had Freetube 0.18 installed with the same issues on both my installations (linux laptop and windows pc). Then I followed the suggestion to install the nightly build with only a minor easing effect of the buffer problem. Then, when 0.19 got released, I installed it, as previously, and "replaced" the existing 0.18 installation. Guess what, the result was no change to the buffering situation.
Then I asked myself, why I never had to port / copy my configuration, subscriptions nor history to the nightly build I installed aside the previous 0.18 regular installation. So I searched my filesystem and discovered, that not only is there a freetube installation directory but also one that contains all configuration and user defined content in the following path: C:\Users\%USERNAME%\AppData\Roaming\FreeTube
So I decided to remove all installations of freetube on my machine, reinstalled freetube the regular way again and made sure my "test" was consistend. The buffering issue was still present. After that I again, deinstalled freetube.
After that I also removed all Freetube data inside my appdata. After that I installed the 0.19 again and imported my previously made exports of my subscriptions, history and so on.

After this steps everything regarding the buffering has been resolved for my case. Maybe this helps someone. I have the feeling that some old configuration, sticking inside the appdata caused an old cause of the issue to persist even if you installed or updated to a newer version of freetube.

If it helps; My first freetube installation was 0.11.2 if Im not wrong. Also I made copys of all the old directorys before cleaning/deleting them.
